spec/mem/tree_borrows/memory.md

```diff
@@ -50,6 +50,8 @@ type TreeBorrowsAllocation = Allocation<TreeBorrowsProvenance, TreeBorrowsAlloca
 ```rust
 pub struct TreeBorrowsMemory<T: Target> {
     allocations: List<TreeBorrowsAllocation>,
+    /// Map from active call IDs to their protecting nodes.
+    active_calls: Map<CallId, List<TreeBorrowsProvenance>>,
```

Comment:
There seems to be a state invariant here: a `TreeBorrowsProvenance` is in the list for some `CallId` if and only if it is currently (weakly or strongly) protected.

That should be documented. Also, I wonder if we can avoid that redundancy...